### PR TITLE
New mobx demo based on the official one

### PR DESCRIPTION
**Pull request by mcallisto**
Refactoring of [MobX getting started example](https://mobx.js.org/getting-started)

### CODE REVIEW

File: react-mobx/src/main/scala/demo/TodoView.scala [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+  val component: FunctionalComponent[Props] = FunctionalComponent[Props] {
+    case Props(todo, toggle, rename) =>
+      val onRename = () =>
+        window.prompt("Task name", todo.task) match {
+          case e if e.isEmpty => ()
+          case task           => rename(task)
+        }
+
+      val onToggleCompleted = () => {
```
Comment (oyvindberg) [inline react-mobx/src/main/scala/demo/TodoView.scala]: inline this? or rename in the pattern match

File: react-mobx/src/main/scala/demo/TodoList.scala [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        store.increasePending()
+        window.setTimeout(() => {
+          store.addTodo("Random Todo " + Math.random())
+          store.decreasePending()
+        }, 2000)
+        ()
+      }
+
+      val todoViews = {
+        val ts = store.todos.get().l
```
Comment (oyvindberg) [inline react-mobx/src/main/scala/demo/TodoList.scala]: `.l`? What's that can you remove it?

File: react-mobx/src/main/scala/demo/ObservingFC.scala [hunk 3]
```diff
@@ -1,11 +0,0 @@
-package slinky.core
```
Comment (oyvindberg) [inline react-mobx/src/main/scala/demo/ObservingFC.scala:1]: I had forgot I had implemented this twice, and this is way shorter. should we keep this implementation and the other name?

**Review by oyvindberg — APPROVED** [review]
I expect you can trim the dependency list in `build.sbt` as well a bit?

**Comment by oyvindberg** [comment]
I think you can merge right? just merge whenever you're happy :)

File: react-mobx/src/main/scala/demo/ObservingFC.scala (hunk 3 above)
Comment (mcallisto) [reply react-mobx/src/main/scala/demo/ObservingFC.scala:1]: So should I use this leaner implementation also to substitute https://github.com/ScalablyTyped/mobx-slinky/blob/master/src/main/scala/slinky/core/ObserverComponent.scala?

Comment (oyvindberg) [reply react-mobx/src/main/scala/demo/ObservingFC.scala:1]: well yeah I suppose so. It's not very important though, they should work exactly the same

File: react-mobx/src/main/scala/demo/TodoView.scala (hunk 1 above)
Comment (mcallisto) [reply react-mobx/src/main/scala/demo/TodoView.scala]: Solved with https://github.com/ScalablyTyped/SlinkyDemos/pull/21/commits/31306a27cae073ac9c56f7617344b43a1eddc8c4

File: react-mobx/src/main/scala/demo/TodoList.scala (hunk 2 above)
Comment (mcallisto) [reply react-mobx/src/main/scala/demo/TodoList.scala]: Solved with https://github.com/ScalablyTyped/SlinkyDemos/pull/21/commits/31306a27cae073ac9c56f7617344b43a1eddc8c4

**Pull request merged**